components/raftstore/src/store/fsm/store.rs

```diff
@@ -163,6 +166,64 @@ impl StoreMeta {
         let reader = self.readers.get_mut(&region.get_id()).unwrap();
         peer.set_region(host, reader, region);
     }
+
+    /// Update damaged ranges and return true if overlap exists.
+    ///
+    /// Condition:
+    /// end_key > file.smallestkey
+    /// start_key <= file.largestkey
+    pub fn update_overlap_damaged_ranges(&mut self, fname: &str, start: &[u8], end: &[u8]) -> bool {
+        for (_, id) in self
+            .region_ranges
+            .range((Excluded(start.to_owned()), Unbounded::<Vec<u8>>))
+        {
+            let region = &self.regions[id];
+            if keys::enc_start_key(region).as_slice() <= end {
+                match self.damaged_ranges.entry(fname.to_owned()) {
```

Comment:
use `try_insert`

Reply:
seem that it's a unstable feature and can you explain why it's better than `.entry()`?

Reply:
Can use `if let`